source/lac/sparse_direct.cc

```diff
@@ -268,24 +293,45 @@ SparseDirectUMFPACK::factorize(const Matrix &matrix)
   sort_arrays(matrix);
 
   int status;
-  status = umfpack_dl_symbolic(N,
-                               N,
-                               Ap.data(),
-                               Ai.data(),
-                               Ax.data(),
-                               &symbolic_decomposition,
-                               control.data(),
-                               nullptr);
+  if (numbers::NumberTraits<number>::is_complex == false)
```

Comment:
I'll approve this PR without this change (this is just a minor style request), but would you mind putting the positive check first here and below?

Reply:
Well, the way I thought about it is that this is actually the default case: The real-valued case. We just don't have a `NumberTraits::is_real` trait.